JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -121,71 +113,64 @@ fun HomeRoute(
-                    onInteractWithList = onInteractWithFeed,
-                    onInteractWithDetail = onInteractWithArticleDetails,
-                    openDrawer = openDrawer,
-                    navigateToInterests = navigateToInterests,
-                    homeListLazyListState = homeListLazyListState,
-                    articleDetailLazyListStates = articleDetailLazyListStates,
-                    scaffoldState = scaffoldState,
-                    modifier = screenModifier
-                )
-            }
-            HomeScreenType.Feed -> {
-                HomeFeedScreen(
-                    uiState = uiState,
-                    showNavRail = showNavRail,
-                    onToggleFavorite = onToggleFavorite,
-                    onSelectPost = onSelectPost,
-                    onRefreshPosts = onRefreshPosts,
-                    onErrorDismiss = onErrorDismiss,
-                    openDrawer = openDrawer,
-                    navigateToInterests = navigateToInterests,
-                    homeListLazyListState = homeListLazyListState,
-                    scaffoldState = scaffoldState,
-                    modifier = screenModifier
-                )
-            }
-            HomeScreenType.ArticleDetails -> {
-                // Guaranteed by above condition for home screen type
-                check(uiState is HomeUiState.HasPosts)
+    // Modifier to be applied to the all HomeScreen types
+    val screenModifier = Modifier.homeScreenPadding(showTopAppBar)
+    val homeScreenType = getHomeScreenType(showTopAppBar, uiState)
+    when (homeScreenType) {
+        HomeScreenType.FeedWithArticleDetails -> {
+            HomeFeedWithArticleDetailsScreen(
+                uiState = uiState,
+                showTopAppBar = showTopAppBar,
+                onToggleFavorite = onToggleFavorite,
+                onSelectPost = onSelectPost,
+                onRefreshPosts = onRefreshPosts,
+                onErrorDismiss = onErrorDismiss,
+                onInteractWithList = onInteractWithFeed,
+                onInteractWithDetail = onInteractWithArticleDetails,
+                openDrawer = openDrawer,
+                homeListLazyListState = homeListLazyListState,
+                articleDetailLazyListStates = articleDetailLazyListStates,
+                scaffoldState = scaffoldState,
+                modifier = screenModifier
+            )
+        }
+        HomeScreenType.Feed -> {
+            HomeFeedScreen(
+                uiState = uiState,
+                showTopAppBar = showTopAppBar,
+                onToggleFavorite = onToggleFavorite,
+                onSelectPost = onSelectPost,
+                onRefreshPosts = onRefreshPosts,
+                onErrorDismiss = onErrorDismiss,
+                openDrawer = openDrawer,
+                homeListLazyListState = homeListLazyListState,
+                scaffoldState = scaffoldState,
+                modifier = screenModifier
```

Comment:
When just the feed is showing with no nav bar, the article list can be obscured by the navigation bar:

![device-2021-09-17-141824](https://user-images.githubusercontent.com/4217560/133854676-de37df80-2666-494e-9433-61352ed69acb.png)

This also ideally would use `contentPadding` for nice edge-to-edge support.

Reply:
Done, thanks!